sources/common/src/VCDTracer.cpp

```diff
@@ -61,9 +64,20 @@ void TRACER::VCDTracer::GenerateHeader()
 
 void TRACER::VCDTracer::GenerateBasicInformation()
 {
-    DumpLine("$date December 8, 2014 14:15:00");
+    auto now = std::chrono::system_clock::now();
+    auto time = std::chrono::system_clock::to_time_t(now);
+    #ifdef WIN32
+#pragma warning(disable : 4996)
+    #endif
+    char *pTimeStr = ctime(&time);
```

Comment:
Use `std::put_time` to get rid of this macro mess.

Reply:
You also need to use `std::localtime` to get `std::tm` object.
I suggest using `"%F %T"` as format to `std::put_time`.

Reply:
It's nonsense. `std::localtime` is not deprecated in any way. C++ is inherently unsafe, every pointer operation should trigger such general warning. :-/ I know how `std::localtime` may use static variable for return value but we are not doing anything unsafe here. Same with pointer arithmetic - you could _possibly_ fail miserably.

Please at least put that horrible mess (with `std::put_time` or old version) in separate function.